gradle-plugins/model-compiler/src/test/java/io/spine/gradle/compiler/lookup/valrule/ValidationRulesLookupPluginShould.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.gradle.compiler.lookup.valrule;
+
+import io.spine.gradle.compiler.GradleProject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Properties;
+
+import static io.spine.gradle.TaskName.FIND_VALIDATION_RULES;
+import static io.spine.gradle.compiler.Extension.getDefaultMainGenResDir;
+import static io.spine.gradle.compiler.lookup.valrule.ValidationRulesFinder.PROTO_TYPE_SEPARATOR;
+import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Dmytro Grankin
+ */
+public class ValidationRulesLookupPluginShould {
+
+    private static final String PROJECT_NAME = "validation-rules-lookup-plugin-test";
+
+    private static final String PROTO_FILE_PACKAGE = "test.valrule";
+    private static final String OUTER_MESSAGE_TYPE = "Outer";
+    private static final String VALIDATION_RULE_TYPE = "ValidationRule";
+    private static final String VALIDATION_TARGET = PROTO_FILE_PACKAGE + PROTO_TYPE_SEPARATOR +
+            OUTER_MESSAGE_TYPE + PROTO_TYPE_SEPARATOR + "field_name";
+    private static final List<String> NESTED_VALIDATION_RULE_PROTO =
+            Arrays.asList("syntax = \"proto3\";",
+                          "package " + PROTO_FILE_PACKAGE + ';',
+                          "import \"spine/options.proto\";",
+
+                          "message " + OUTER_MESSAGE_TYPE + " {",
+
+                          "message " + VALIDATION_RULE_TYPE + " {",
+                          "option (validation_of) = \"" + VALIDATION_TARGET + "\";",
+                          "}",
+
```

Comment:
We need to know a significant part of the file content. So, the approach with the file does not make much sense, because it requires parsing of the file, that causes the code complication.